tests/unit/test_flist.py

```diff
@@ -82,30 +87,61 @@ def test_api_foldLeft(self):
     self.assertEqual(120.0, alist.foldLeft(1.0, lambda a, e: a * e))
 
   def test_api_foldLeft_with_short_stop(self):
+    """Testing API Cons.foldLeft with arg short_stop_func."""
     alist = fl(1, 2, 3, 4, 5)
-
-    result = alist.foldLeft(
-        0, lambda a, b: a + b,
-        short_stop_func=lambda v: v > 3)
+    f = lambda a, b: a + b # noqa
+    # test corner case where the func stops at identity
+    result = alist.foldLeft(0, f, short_stop_func=lambda v: v > -1)
+    self.assertIsInstance(result, Nil)
 
     # The folding operation will stop at element 4.
     # So we have folding result as 0 + 1 + 2 + 3 = 6
+    result = alist.foldLeft(0, f, short_stop_func=lambda v: v > 3)
     self.assertEqual(result, 6)
 
+  def test_api_foldReft(self):
```

Comment:
According to the comment, maybe the name of unit test case should be `test_api_foldRight`?